core/src/test/scala/unit/kafka/log/LogCleanerTest.scala

```diff
@@ -363,13 +363,13 @@ class LogCleanerTest {
     assertEquals(List(2, 1, 3), LogTest.keysInLog(log))
     assertEquals(List(3, 4, 5, 6, 7, 8), offsetsInLog(log))
 
-    // delete horizon forced to 0 to verify marker is not removed early
-    dirtyOffset = cleaner.doClean(LogToClean(tp, log, dirtyOffset, log.activeSegment.baseOffset), currentTime = 0L)._1
+    // delete horizon forced to smaller timestamp to verify marker is not removed early
+    dirtyOffset = cleaner.doClean(LogToClean(tp, log, dirtyOffset, log.activeSegment.baseOffset), currentTime = largeDeleteHorizon)._1
```

Comment:
This is still a bit confusing. Could we define 2 vals, beforeDeleteHorizon and afterDeleteionHorizon? The former takes Long.MaxValue - tombstoneRetentionMs - 1 and the latter takes Long.MaxValue. The comment can be changed to sth like "current time is still before deleteHorizon". It would be useful to do this consistently across other tests.